game-core/src/main/java/games/strategy/triplea/attachments/UnitAttachment.java

```diff
@@ -1479,9 +1480,15 @@ private int getMovement() {
   }
 
   public int getMovement(final GamePlayer player) {
+    final int nonCombatMoveBonus =
+        GameStepPropertiesHelper.isNonCombatMove(getData(), true)
+            ? TechAbilityAttachment.getNonCombatMovementBonus(
+                (UnitType) this.getAttachedTo(), player, getData())
+            : 0;
     return Math.max(
         0,
         movement
+            + nonCombatMoveBonus
```

Comment:
Do I understand right that the bonus is not additive? IE: if the tech says movement is '1', it's not +1 but actually set to a movement of one? If I understand that right, could you walk me through the reasoning why the bonus should not be additive?